src/scheduling/heuristics/initialize.py

```diff
@@ -0,0 +1,339 @@
+) -> list[int]:
+    if end - start <= max_qpu_size:
+        return []
+    possible_cuts = [_calulate_cut(counts, cut) for cut in range(start + 1, end - 2)]
+    best_cut = min(possible_cuts, key=lambda cut: cut[1])[0]
+    partitions = (
+        [best_cut]
+        + _find_cuts(counts, start, best_cut, max_qpu_size)
+        + _find_cuts(counts, best_cut + 1, end, max_qpu_size)
+    )
+
+    return partitions
+
+
+def _calulate_cut(counts: Counter[tuple[int, int]], cut: int) -> tuple[int, int]:
+    left = sum(
+        count for (first, second), count in counts.items() if first <= cut < second
+    )
+    right = sum(
+        count for (first, second), count in counts.items() if second <= cut < first
+    )
+    return cut, left + right
+
+
+def _choice_partitioning(
+    circuits: list[QuantumCircuit],
+    accelerators: list[Accelerator],
+    **kwargs,
+) -> list[list[int]]:
+    partitions = []
+    qpu_sizes = [acc.qubits for acc in accelerators]
+    circuit_sizes = [circ.num_qubits for circ in circuits]
+    for circuit_size in sorted(circuit_sizes, reverse=True):
+        partition = []
+        remaining_size = circuit_size
+        while remaining_size > 0:
+            qpu = np.random.choice(qpu_sizes)
+            take_qubits = min(remaining_size, qpu)
+            if remaining_size - take_qubits == 1:
+                # We can't have a partition of size 1
+                # So in this case we take one qubit less to leave a partition of two
+                take_qubits -= 1
+            partition.append(take_qubits)
+            remaining_size -= take_qubits
+        partitions.append(partition)
+    return partitions
+
+
+def _random_partitioning(
+    circuits: list[QuantumCircuit],
+    accelerators: list[Accelerator],
+    **kwargs,
+) -> list[list[int]]:
+    partitions = []
+    max_qpu_size = max(acc.qubits for acc in accelerators) + 1
+    circuit_sizes = [circ.num_qubits for circ in circuits]
+    for circuit_size in sorted(circuit_sizes, reverse=True):
+        partition = []
+        remaining_size = circuit_size
+        if circuit_size <= 3:
+            partitions.append([circuit_size])
+            continue
+        while remaining_size > 0:
+            qpu = np.random.randint(2, max_qpu_size)
+            take_qubits = min(remaining_size, qpu)
+            if remaining_size - take_qubits == 1:
+                # We can't have a partition of size 1
+                # So in this case we take one qubit less to leave a partition of two
+                take_qubits -= 1
+            partition.append(take_qubits)
+            remaining_size -= take_qubits
+        partitions.append(partition)
+    return partitions
+
+
+def _fixed_partitioning(
+    circuits: list[QuantumCircuit],
+    accelerators: list[Accelerator],
+    **kwargs,
+) -> list[list[int]]:
+    if "partition_size" not in kwargs:
+        partition_size = 10
+    else:
+        partition_size = kwargs["partition_size"]
```

Comment:
```suggestion
    partition_size = kwargs.pop("partition_size", 10)
```
or equivalently with `get` if you don't want to mutate the kwargs